### ISSUE TITLE
Set up to train/test on pre-processed UMAP embeddings?

### ISSUE THREAD

**Issue by kxchiu** [MaartenGr/BERTopic#1114]
Hi Maarten,

First off, thank you for the elegant model. It works really well on a project that I'm working on as I followed the examples to train on text docs and embeddings.

I want to know if there's a proper set-up to train/test with pre-processed UMAP embeddings -- e.g. someone else on my team would preprocess the text documents to generate the HF embeddings and run through UMAP to get the dimensionally reduced UMAP embeddings, and then I could try to fit BERTopic on to the UMAP embeddings.

I checked the documentation page "Embeddings" and "Dimensionality Reduction" and found that we could pass in pre-trained sentence transformers embeddings when fitting, like:
```
topics, probs = topic_model.fit_transform(docs, embeddings)
```
and skip dimensionality reduction by passing in the BaseDimensionalityReduction():
```
# Fit BERTopic without actually performing any dimensionality reduction
empty_dimensionality_model = BaseDimensionalityReduction()
topic_model = BERTopic(umap_model=empty_dimensionality_model)
```

I tried to combine the two approaches. However, the topics generated seemed to be less steered to the pre-labeled categories, and the performance of BERTopic doesn't seem to hold up when I had the text and embeddings as the input. The code snippet that I used was as the following:
```
# Fit BERTopic without actually performing any dimensionality reduction
empty_dimensionality_model = BaseDimensionalityReduction()
topic_model = BERTopic(umap_model=empty_dimensionality_model)

# Train with semi-supervised topic modeling (about 37% matching ground truth labels)
topics, probs = topic_model.fit_transform(train_docs, 
    train_umap_embeddings, 
    y=categories)

# Test (36~38% accuracy)
test_topics, test_probs = topic_model.transform(test_docs,
    test_umap_embeddings)
```
By accuracy, I was measuring the ground truth category label to the generated topics, which are post-labeled by the prominent category in the topic.

Do you have any suggestions on how BERTopic may work with taking preprocessed UMAP embeddings as the input embedding?

**Comment by MaartenGr** [comment]
If you have ground truth labels for your data, I would advise going with [fully-supervised topic modeling](https://maartengr.github.io/BERTopic/getting_started/supervised/supervised.html) instead. The semi-supervised topic modeling is merely steering towards the creation of the supplied categories but also creates its owns and is stubborn enough to ignore those if they do not fit.

**Comment by MaartenGr** [comment]
Also, HDBSCAN is not really the best when it comes to predicting the topics/clusters of unseen documents. For that, a fully-supervised model would work much better. You could even try out k-Means as that model is a bit more constrained and therefore likely better for your use case.

**Comment by kxchiu** [comment]
@MaartenGr your suggestion works exactly how we expected. Thanks!
